WPFHospitalEditor/AdditionalInformation.xaml.cs

```diff
@@ -0,0 +1,233 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Windows;
+using System.Windows.Controls;
+using System.Windows.Data;
+using System.Windows.Documents;
+using System.Windows.Input;
+using System.Windows.Media;
+using System.Windows.Media.Imaging;
+using System.Windows.Shapes;
+using WPFHospitalEditor.MapObjectModel;
+
+namespace WPFHospitalEditor
+{
+    /// <summary>
+    /// Interaction logic for AdditionalInformation.xaml
+    /// </summary>
+    public partial class AdditionalInformation : Window
+    {
+        private MapObject mapObject;
+        private List<string> labelContent = new List<string>();
+        private List<string> value = new List<string>();
+        private List<Label> labels = new List<Label>();
+        private List<TextBox> textBoxes = new List<TextBox>();
+        private TextBox name = new TextBox();
+
+        public AdditionalInformation(MapObject mapObject)
+        {
+            InitializeComponent();
+            this.mapObject = mapObject;
+            string[] contentRows = mapObject.Description.Split(";");
+            this.Height = (contentRows.Length + 2) * 50 + 30;
+            CreateDynamicWPFGrid(mapObject, contentRows);   
+        }
+
+        private void Close_Click(object sender, RoutedEventArgs e)
+        {
+            Close();
+        }
+
+        private void Done_Click(object sender, RoutedEventArgs e)
+        {
+            mapObject.Description = "";
+
+            for(int i = 0; i < labels.Count; i++)
+            {
+                mapObject.Description += labels[i].Content;
+                mapObject.Description += "=";
+                mapObject.Description += textBoxes[i].Text;
+                mapObject.Description += ";";
+            }
+
+            int lenght = mapObject.Description.Length;
+            mapObject.Description.Substring(0,lenght - 1);
+
+            mapObject.name.Text = this.name.Text;
+            
+            Close();
+        }
+
+        private void CreateDynamicWPFGrid(MapObject mapObject, string[] contentRows)
+
+        {
+            Grid DynamicGrid = new Grid();
+            var bc = new BrushConverter();
+            DynamicGrid.Background = (Brush)bc.ConvertFrom("#FFC6F5F8");
+
+            createColumns(DynamicGrid);
+
+            createRows(contentRows, DynamicGrid);
+
+            addMapObjectName(mapObject, bc, DynamicGrid);
+
+            createRowContent(contentRows, DynamicGrid);
```

Comment:
DynamicGrid object should be a field in class AdditionalInformation, in this way we will decrease number of parameters of these functions.